kobo/apps/stripe/views.py

```diff
@@ -308,42 +307,51 @@ def generate_portal_link(user, organization_id, price):
             if not len(all_configs):
                 return Response({'error': "Missing Stripe billing configuration."}, status=status.HTTP_502_BAD_GATEWAY)
 
-            # get the portal configuration that lets us switch to the provided price
-            current_config = next(
-                (config for config in all_configs if (
-                        config['active'] and
-                        config['livemode'] == settings.STRIPE_LIVE_MODE and
-                        config['metadata'].get('portal_price', None) == price.id
-                )), None
-            )
+            if price.product.metadata['product_type'] == 'addon':
+                """
+                Recurring add-ons aren't included in the default billing configuration.
+                This lets us hide them as an 'upgrade' option for paid plan users.
+                Here, we try getting the portal configuration that lets us switch to the provided price.
+                """
+                current_config = next(
+                    (config for config in all_configs if (
+                            config['active'] and
+                            config['livemode'] == settings.STRIPE_LIVE_MODE and
+                            config['metadata'].get('portal_price', None) == price.id
```

Comment:
Is `config['metadata'].get('portal_price', None)` always an integer? Ex: `'2' != 2`

Reply:
Stripe metadata values/keys are [always strings](https://stripe.com/docs/api/metadata), and `price.id` should be validated by the `CustomerPortalSerializer`. Let me know if I'm wrong, as is often the case 😅 The expected value of `price.id` would be `price_[some alphanumeric string]`, although someone messing around in the database could put in some other (string) value.